clippy_lints/src/arc_with_non_send_sync.rs

```diff
@@ -61,19 +63,25 @@ impl<'tcx> LateLintPass<'tcx> for ArcWithNonSendSync {
                 cx,
                 ARC_WITH_NON_SEND_SYNC,
                 expr.span,
-                "usage of an `Arc` that is not `Send` or `Sync`",
+                "usage of an `Arc` that is not `Send` and `Sync`",
                 |diag| {
                     with_forced_trimmed_paths!({
+                        diag.note(format!("`Arc<{arg_ty}>` is not `Send` and `Sync` as"));
+
                         if !is_send {
                             diag.note(format!("the trait `Send` is not implemented for `{arg_ty}`"));
                         }
                         if !is_sync {
                             diag.note(format!("the trait `Sync` is not implemented for `{arg_ty}`"));
                         }
```

Comment:
Some list formatting here would be very appreciated for legibility.

```suggestion
                        diag.note(format!("`Arc<{arg_ty}>` is not `Send` and `Sync` as:"));

                        if !is_send {
                            diag.note(format!("- the trait `Send` is not implemented for `{arg_ty}`"));
                        }
                        if !is_sync {
                            diag.note(format!("- the trait `Sync` is not implemented for `{arg_ty}`"));
                        }
```